lib/experimental/html/class-wp-html-walker.php

```diff
@@ -273,25 +273,72 @@ public function next_tag( $query = null ) {
 	/**
 	 * Skips the contents of the title and textarea tags until an appropriate
 	 * tag closer is found:
+	 *
 	 * https://html.spec.whatwg.org/multipage/parsing.html#rcdata-state
 	 *
 	 * @param string $tag_name – the rcdata tag closer to look for.
 	 * @since 6.1.0
 	 */
 	private function skip_rcdata( $tag_name ) {
+		$html       = $this->html;
+		$doc_length = strlen( $html );
+		$tag_length = strlen( $tag_name );
+
+		$at = $this->parsed_bytes;
+
 		while ( true ) {
-			// Find a potential tag closer.
-			$closer = "</{$tag_name}";
-			$at     = strpos( $this->html, $closer, $this->parsed_bytes );
-			if ( false === $at ) {
+			$at = strpos( $this->html, '<', $at );
+
+			// If we have no possible tag closer then fail.
+			if ( false === $at || ( $at + $tag_length ) > $doc_length ) {
+				$this->parsed_bytes = $doc_length;
 				return;
 			}
-			// Confirm the initial </tag sequence is followed by a terminating character.
-			$this->parsed_bytes = $at + strlen( $closer );
-			if ( 0 === strspn( $this->html, " \t\f\r\n/>", $this->parsed_bytes ) ) {
+
+			$at += 1;
+
+			// We need a closing tag to break out of here.
+			if ( '/' !== $html[ $at++ ] ) {
 				continue;
 			}
+
+			/*
+			 * We have to find a case-insensitive match to the tag name.
+			 * Note also that since tag names are limited to US-ASCII
+			 * characters we can ignore any kind of Unicode normalizing
+			 * forms when comparing. If we get a non-ASCII character it
+			 * will never be a match.
+			 */
+			for ( $i = 0; $i < $tag_length; $i++ ) {
+				$tag_char  = $tag_name[ $i ];
+				$html_char = $html[ $at + $i ];
+
+				if ( $html_char !== $tag_char && strtolower( $html_char ) !== strtolower( $tag_char ) ) {
+					$at += $i;
+					continue 2;
+				}
+			}
+
+			$at += $tag_length;
+			$this->parsed_bytes = $at;
+
+			/*
+			 * Ensure we terminate the tag name, otherwise we might,
+			 * for example, accidentally match the sequence
+			 * "</textarearug>" for "</textarea>".
+			 */
+			$c = $html[ $at++ ];
+			if ( ' ' !== $c && "\t" !== $c && "\r" !== $c && "\n" !== $c && '/' !== $c && '>' !== $c ) {
+				continue;
+			}
+
 			$this->skip_tag_closer_attributes();
+			$at = $this->parsed_bytes;
```

Comment:
`$this->skip_whitespace()` could help handle a closer such as `</textarea >`. On the other hand, attributes parser could be a better way for that, especially in cases such as `</textarea / / / / >`